pypechain/core/utilities.py

```diff
@@ -70,6 +70,9 @@ def dataclass_to_tuple(instance: Any) -> Any:
     Any
         either a tuple or the orginial value
     """
+    if isinstance(instance, (list, tuple)):
```

Comment:
return is now a tuple or the original value or a list of tuples. would it make sense to return a tuple here? or maybe better is type(instance)([dataclass_to_tuple(x) for x in instance]) so it inherits the arg type